### ISSUE TITLE
Provide a proper dev environment out of the box

### ISSUE THREAD

**Issue by philippejadin** [agorakit/agorakit#492]
I'm still struggling (when I reinstall) to quickly get up and running to work on Agorakit. As the initial dev of the project, this is problematic :-)

I want : 
- the correct php version & extensions installed
- with composer
- mariadb
- shell access
- env key setted up
- db migrated
- shell access
- direct access to project files as well

So it seems that Docker is the recommended solution. I had some success using Laravel Sail (which uses Docker under the hood), but it is a bit too bloated imvho.

Work happens here : https://github.com/agorakit/agorakit/tree/docker-for-dev/docker/dev

So first, is Docker the best way to approach this?

Alternatives found : 
- https://docs.lando.dev/plugins/laravel/ (it adds one more layer on docker it seems, so is it worth it ?)

Podman seems also a better container engine (better in the sense of really opensource)

**Comment by philippejadin** [comment]
@linc do you have input on this? it is an endless battle for me :-)

**Comment by linc** [comment]
Yeah, if the goal is a container for localhost dev (not production/deployment), I think I’ll have a solution soon. My work is currently in the process of evaluating a new solution to this, and we’re going to choose something open source (and we’re also in PHP), so I hope I can do a 1:1 implementation here. We’re currently using “devilbox” (which uses Docker undermeath) but it’s been unmaintained for years so it’s time to move on.

**Comment by philippejadin** [comment]
I came across devilphp, and a lot of other solutions. After a while I'm thinking, aren't those another unneeded abstraction layer? I'm somewhat happy with the docker build file in the docker branch. I'll merge it when it will be good enough for me, while being very open to anything even more simple 🙂

**Comment by linc** [comment]
I don't even use a container-level abstraction, I use this 10,000-word guide I wrote ;) https://lincolnwebs.com/php-localhost/ — I run all the processes natively to conserve performance (and because I learned how before Docker existed  & want to stay sharp, so why not?)

However if you want to make it easy for _anyone_ to approach setting up their localhost quickly, some sort of automation on top of Docker tends to help people get moving faster.

That said, I really don't have strong feelings about this. Happy to do my best to help in whichever direction you prefer. I only suggested the wrapper because I have folks smarter than me who will be working on it anyway (it's their preferred solution as a team).

**Comment by philippejadin** [comment]
That's some intense & impressive guide :-)

I guess it's a mater of personal preference, so chasing the best solution is probably meaningless. I'd be happy to discover what your colleagues propose.


In the meantime I think I'll keep using my light docker based setup I just manage to make work, just like you :-)

**Comment by philippejadin** [comment]
I consider the latest iteration of docker setup working in /docker/dev
The [readme](https://github.com/agorakit/agorakit/blob/main/docker/dev/README.md) provides enough background imho.

Closing this for now, but more documentation is needed in the documentation repository.

**Comment by zerline** [comment]
Hi!
I commit to writing some doc here.

**Comment by philippejadin** [comment]
@zerline great! You can base the doc on this temporary document : https://github.com/agorakit/agorakit/blob/main/docker/dev/README.md

**Comment by zerline** [comment]
Great!
While testing these docker images:

- everything installs perfectly
- I can access all 3 services with https://localhost/, https://localhost/8080 and https://localhost/8025, just as described
- the verification email is not sent, apparently because the SMTP port is not properly configured ; here is the complete message I get:  `[15:01:36] LOG.error: Connection could not be established with host "mailpit:1025": stream_socket_client(): php_network_getaddresses: getaddrinfo for mailpit failed: Name or service not known {
    "userId": 1,
    "exception": {}
}
`

**Comment by philippejadin** [comment]
The culprit is probably here : https://github.com/agorakit/agorakit/blob/ab6432727070feaa2508a27677e4c1956b8e2318/.env.dev#L37

I'm not on a computer to check

**Comment by zerline** [comment]
Yes. Therefore, should we directly modify the variable MAIL_HOST in _.env.dev_?

**Comment by philippejadin** [comment]
Yes it seems the most logical way.

Le sam. 7 déc. 2024, 00:52, Odile Bénassy ***@***.***> a
écrit :

**Comment by linc** [comment]
This is working, right? Can this be closed?

**Comment by philippejadin** [comment]
Yes, imho it works fine and is not too ressource intensive to build.

I let it open since it needs doc, but it is done in fact.



Le 29 mars 2025 17:37:07 GMT+01:00, Lincoln Russell ***@***.***> a écrit :
>linc left a comment (agorakit/agorakit#492)
>
>This is working, right? Can this be closed?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/agorakit/agorakit/issues/492#issuecomment-2763667149
>You are receiving this because you modified the open/close state.
>
>Message ID: ***@***.***>
agorakit.org
